xlators/cluster/dht/src/dht-rename.c

```diff
@@ -1385,12 +1392,20 @@ dht_rename_create_links(call_frame_t *frame)
      * unlink the newname we created, we would have effectively lost the
      * file to rename operations. */
     if (dst_hashed != src_hashed && src_cached != dst_hashed) {
-        gf_msg_trace(this->name, 0, "linkfile %s @ %s => %s", local->loc.path,
-                     dst_hashed->name, src_cached->name);
+        dict_t *xattr_new = NULL;
+
+        xattr_new = dict_copy_with_ref(xattr, NULL);
+        DHT_MARK_RENAME_FOP(xattr_new);
+        local->params = dict_ref(xattr_new);
```

Comment:
Instead of taking a new ref we can simply assign `xattr_new` here and remove the unref at line 1408.